animation_nodes/nodes/gpencil/gp_object_output.py

```diff
@@ -0,0 +1,87 @@
+import bpy
+from bpy.props import *
+from ... base_types import AnimationNode, VectorizedSocket
+
+class GPObjectOutputNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GPObjectOutputNode"
+    bl_label = "GP Object Output"
+    errorHandlingType = "EXCEPTION"
+
+    useLayerList: VectorizedSocket.newProperty()
+
+    def create(self):
+        socket = self.newInput("Object", "Object", "object")
+        socket.defaultDrawType = "PROPERTY_ONLY"
+        socket.objectCreationType = "GPENCIL"
+        self.newInput(VectorizedSocket("GPLayer", "useLayerList",
+            ("Layer", "layer"), ("Layers", "layers")))
+        self.newOutput("Object", "Object", "object")
+
+    def getExecutionFunctionName(self):
+        if self.useLayerList:
+            return "execute_LayerList"
+        else:
+            return "execute_Layer"
+
+    def execute_Layer(self, object, layer):
+        if object is None: return None
+        gpencil = self.getObjectData(object)
+
+        if layer is None: return object
+        self.setLayerData(gpencil, layer)
+        gpencil.layers.active.frames.update()
+        return object
+
+    def execute_LayerList(self, object, layers):
+        if object is None: None
+        gpencil = self.getObjectData(object)
+
+        if len(layers) == 0: return object
+        for layer in layers:
+            self.setLayerData(gpencil, layer)
+        gpencil.layers.active.frames.update()
+        return object
+
+    def setLayerData(self, gpencil, layer):
+        gpencilLayer = self.getLayer(gpencil, layer)
+        gpencilLayer.clear()
```

Comment:
Clear inside the `getLayer` function.

I wounder if we should clear the whole object data. So the layers of the object will be only those we input.

Reply:
I have tried that, But If we clear the whole data then it is not possible to add manual layers and draw on them. That is the reason I only clear the layers which are added in the AN.

Reply:
Why is this the case?

Reply:
If we clear the data of the object it clears everything that means all layers of the object.

Reply:
Ok, but that's what I am after. I would the expect the object to have only the layers that I gave it. It is like the Spline Object Output node. The exiting splines are removed and only the ones I gave it are there.

Reply:
This will remove the artist's control over the GP objects from AN. The reason I have reworked the whole GP nodes to add many frames and layers through the AN as well as a manual which will make these nodes more powerful. Here is a simple example, a tree trunk is drawn manually but leaves movement is controlled by AN. 

We can add a button in the GP Output node to clear all the layers of the GP object if someone need to clear them.

Reply:
On the other hand, one will not be able to remove existing layers. To preserve existing layers, one might read the layers from the object, append one's layers, then write them.

Maybe we can add an option in the advanced node settings to control this?

Reply:
Here is my plan:
1. By default, the GP Output node clears all layers.
2. If someone wants to add a custom layer(s) then it will enable the _Append_ option in the advanced node settings. Then, GP Output node does not clear all the layers, only the layers from the AN.

What do you think?

Reply:
Looks good. :+1:

Reply:
Okay, I'll make the changes.

Reply:
Added the option for appending custom layers.